### PR TITLE
provisioner/chef: tweaking the tests

### PR DESCRIPTION
**Pull request by svanharmelen**
We should only test the uploaded file, and make sure we also test cases were we do not specify a secrets file

### CODE REVIEW

File: helper/schema/valuetype_string.go [hunk 1]
```diff
@@ -9,7 +9,7 @@ const _ValueType_name = "TypeInvalidTypeBoolTypeIntTypeFloatTypeStringTypeListTy
 var _ValueType_index = [...]uint8{0, 11, 19, 26, 35, 45, 53, 60, 67, 77}
 
 func (i ValueType) String() string {
-	if i < 0 || i >= ValueType(len(_ValueType_index)-1) {
+	if i < 0 || i+1 >= ValueType(len(_ValueType_index)) {
```
Comment (phinze) [inline helper/schema/valuetype_string.go]: @svanharmelen were these this `stringer` changes? seems like this diff is unrelated to the test change... did it accidentally sneak in?

Comment (svanharmelen) [inline helper/schema/valuetype_string.go]: Yes it accidentally sneaked in, my bad! Will revert it in a few (if not done already). This is caused by doing these tweaks on an out of date macbook (so with an older stringer package)...
